File: Implimentations/AGE R1/Anarian Game Engine/Anarian Game Engine.Shared/SceneManager.cpp
#include "pch.h"
#include "SceneManager.h"
using namespace Anarian;

SceneManager::SceneManager()
{
}


SceneManager::~SceneManager()
{
}

IScene* SceneManager::CurrentScene()
{
	return m_currentScene;
}
void SceneManager::CurrentScene(IScene* scene)
{
	delete m_currentScene;
	m_currentScene = scene;
}

File: EmpiresOfTheIV/EmpiresOfTheIV/EmpiresOfTheIV.Shared/Anarian/SceneManager.cs
using System;
using System.Collections.Generic;
using System.Text;
using Anarian.Interfaces;

namespace Anarian
{
    public class SceneManager
    {
        #region Singleton
        static SceneManager m_instance;
        public static SceneManager Instance
        {
            get
            {
                if (m_instance == null) m_instance = new SceneManager();
                return m_instance;
            }
            set { }
        }
        #endregion

        private SceneManager()
        {
            CurrentScene = null;
        }

        public IScene CurrentScene { get; set; }
    }
}
